fidesctl/src/fidesctl/core/generate_dataset.py

```diff
@@ -82,6 +85,19 @@ def get_mssql_collections_and_fields(engine: Engine) -> Dict[str, Dict[str, List
     return db_tables
 
 
+def get_db_collections_and_fields(engine: Engine) -> Dict[str, Dict[str, List[str]]]:
+    """
+    Returns a db collections and fields, delegating to a specific engine dialect function
+    """
+    database_ingestion_functions = {
```

Comment:
Yeah that seems like a good idea. It's a little verbose because the type is `Dict[str, Callable[[Engine], Dict[str, Dict[str, List[str]]]]]`

Verified that if one of the functions returns something different then we have a mypy failure:
```
root@d12908839f6b:/fides/fidesctl# mypy
src/fidesctl/core/generate_dataset.py:93: error: Dict entry 0 has incompatible type "str": "Callable[[Engine], Dict[str, Dict[str, List[int]]]]"; expected "str":
"Callable[[Engine], Dict[str, Dict[str, List[str]]]]"  [dict-item]
            "postgresql": get_postgres_collections_and_fields,
            ^
Found 1 error in 1 file (checked 36 source files)
```

Reply:
Personally I think it's worth the ugliness. Thanks for investigating!